src/main/java/com/intel/genomicsdb/GenomicsDBImporter.java

```diff
@@ -244,50 +244,54 @@ private static native String jniGetChromosomeIntervalsForColumnPartition(
      *
-     * @param baseImportConfig basic import configuration
+     * @param parallelImportConfig basic import configuration
      * @throws IOException
      */
-    public static void parallelImport(final BaseImportConfig baseImportConfig) throws IOException, InterruptedException {
-        parallelImport(baseImportConfig, 0);
+    public static void parallelImport(final ParallelImportConfig parallelImportConfig) throws IOException, InterruptedException {
+        parallelImport(parallelImportConfig, 0);
     }
 
     /**
      * Import multiple chromosome interval
      *
-     * @param baseImportConfig basic import configuration
+     * @param parallelImportConfig basic import configuration
      * @param numThreads       number of threads to use
      */
-    public static void parallelImport(final BaseImportConfig baseImportConfig, final int numThreads) throws IOException, InterruptedException {
+    public static void parallelImport(final ParallelImportConfig parallelImportConfig, final int numThreads) throws IOException, InterruptedException {
         //This sorts the list sampleNames if !useSamplesInOrder
         //Why you should use this? If you are writing multiple partitions in different machines,
         //you must have consistent ordering of samples across partitions. If file order is different
         //in different processes, then set useSamplesInOrder to false and let the sort in
         //generateSortedCallSetMap ensure consistent ordering across samples
         GenomicsDBCallsetsMapProto.CallsetMappingPB callsetMappingPB =
                 GenomicsDBImporter.generateSortedCallSetMap(new ArrayList<>(
-                        baseImportConfig.getSampleNameToVcfPath().keySet()), baseImportConfig.isUseSamplesInOrderProvided());
+                        parallelImportConfig.getSampleNameToVcfPath().keySet()),
+                        parallelImportConfig.getImportConfiguration().getGatk4IntegrationParameters().getUseSamplesInOrderProvided());
 
         //Write out callset map if needed
-        if (baseImportConfig.getCallsetOutputFilepath() != null)
-            GenomicsDBImporter.writeCallsetMapJSONFile(baseImportConfig.getCallsetOutputFilepath(), callsetMappingPB);
+        String outputCallsetmapJsonFilePath = parallelImportConfig.getImportConfiguration().getGatk4IntegrationParameters().getOutputCallsetmapJsonFile();
+        if (outputCallsetmapJsonFilePath != null && !outputCallsetmapJsonFilePath.isEmpty())
+            GenomicsDBImporter.writeCallsetMapJSONFile(outputCallsetmapJsonFilePath, callsetMappingPB);
 
         //Write out vidmap if needed
-        if (baseImportConfig.getVidmapOutputFilepath() != null)
-            GenomicsDBImporter.writeVidMapJSONFile(baseImportConfig.getVidmapOutputFilepath(), baseImportConfig.getMergedHeader());
+        String vidmapOutputFilepath = parallelImportConfig.getImportConfiguration().getGatk4IntegrationParameters().getOutputVidmapJsonFile();
```

Comment:
This level of indirection should be fixed once we have a better protobuf definition

Reply:
I'm fine with this for now - this would be changed to something like 
`String vidmapOutputFilepath = parallelImportConfig.getImportConfiguration().getOutputVidmapJsonFile(); `
once [#168](https://github.com/Intel-HLS/GenomicsDB/pull/168) gets merged

Reply:
Yeap, I'm aware of that change but since this PR comes from master I had to use the old protobuf struct :(.